### PR TITLE
Try to fix TORCH_CUDA_ARCH_LIST for PyTorch again

### PR DESCRIPTION
**Pull request by ssnl**
test CI

### CODE REVIEW

**Comment by SsnL** [comment]
yay this passed tests!!!!

File: cmake/public/cuda.cmake [hunk 1]
```diff
@@ -299,28 +299,61 @@ function(caffe2_select_nvcc_arch_flags out_variable)
...
-  elseif(${CUDA_ARCH_NAME} STREQUAL "Volta")
-    set(__cuda_arch_bin "70")
-  elseif(${CUDA_ARCH_NAME} STREQUAL "All")
-    set(__cuda_arch_bin ${Caffe2_known_gpu_archs})
-  elseif(${CUDA_ARCH_NAME} STREQUAL "Manual")
-    set(__cuda_arch_bin ${CUDA_ARCH_BIN})
-    set(__cuda_arch_ptx ${CUDA_ARCH_PTX})
-  elseif(${CUDA_ARCH_NAME} STREQUAL "Auto")
-    caffe2_detect_installed_gpus(__cuda_arch_bin)
+  set(CUDA_ARCH_LIST "")
```
Comment (orionr) [inline cmake/public/cuda.cmake]: This can just be `set(CUDA_ARCH_LIST)`

File: cmake/public/cuda.cmake [hunk 2]
```diff
@@ -299,28 +299,61 @@ function(caffe2_select_nvcc_arch_flags out_variable)
...
+    list(APPEND CUDA_ARCH_LIST ${TORCH_CUDA_ARCH_LIST})
+    message(STATUS "Set CUDA arch from TORCH_CUDA_ARCH_LIST: ${TORCH_CUDA_ARCH_LIST}")
   else()
-    message(FATAL_ERROR "Invalid CUDA_ARCH_NAME")
+    list(APPEND CUDA_ARCH_LIST ${CUDA_ARCH_NAME})
+    message(STATUS "Set CUDA arch from CUDA_ARCH_NAME: ${CUDA_ARCH_NAME}")
   endif()
+  list(REMOVE_DUPLICATES CUDA_ARCH_LIST)
+
+  set(__cuda_arch_bin "")
```
Comment (orionr) [inline cmake/public/cuda.cmake]: These can also be

```
set(__cuda_arch_bin)
set(__cuda_arch_ptx)

**Review by orionr — APPROVED** [review]
Nice! Can you make the minor changes to set before landing and re-run tests? Otherwise looks great.

**Pull request merged**